src/common/hooks/DefaultTopicProgress/DefaultTopicProgress.tsx

```diff
@@ -0,0 +1,126 @@
+import { useEffect, useState, useContext, useMemo } from 'react'
+import log from 'loglevel'
+import { useNavigate } from 'react-router-dom'
+import { AuthContext, SnackbarContext } from '@services'
+import { usePersistedStore, useStore } from '@store'
+import {
+  LearningPathElement,
+  LearningPathElementReturn,
+  LearningPathElementStatus,
+  LearningPathElementStatusReturn,
+  Topic,
+  User
+} from '@core'
+import { SnackbarMessageProps } from '@components'
```

Comment:
Löschen der nicht mehr benötigten Imports und Hinzufügen useCallback

```suggestion
import { useEffect, useState, useContext, useMemo } from 'react'
import log from 'loglevel'
import { useNavigate } from 'react-router-dom'
import { AuthContext, SnackbarContext } from '@services'
import { usePersistedStore, useStore } from '@store'
import {
  LearningPathElement,
  LearningPathElementReturn,
  LearningPathElementStatus,
  LearningPathElementStatusReturn,
  Topic,
  User
} from '@core'
import { SnackbarMessageProps } from '@components'
```